### PR TITLE
Adds the optional publication content type as a feature

### PR DESCRIPTION
**Pull request by ben-hamelin**
## Metadata

| Question | Answer |
|----------|--------|
| Did you use a meaningful pull request title? | yes
| Did you apply meaningful labels to the pull request? | yes
| Did you perform a self review first? | yes
| Documentation reflects changes? | yes
| `CHANGELOG` reflects changes? | yes
| Unit/Functional tests reflect changes? | yes
| Did you perform browser testing? | yes
| Risk level | Low
| Relevant links | JIRA issue, bug reports, security alerts, etc.
https://thinkoomph.jira.com/browse/RIG-53

### CODE REVIEW

File: ecms_base/features/custom/ecms_publications/config/install/field.field.node.publication.field_publication_file_download.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+default_value_callback: ''
+settings:
+  handler: 'default:media'
+  handler_settings:
+    target_bundles:
+      publication_file: publication_file
+    sort:
+      field: _none
+    auto_create: false
+    auto_create_bundle: event_image
```
Comment (pfrilling) [inline ecms_base/features/custom/ecms_publications/config/install/field.field.node.publication.field_publication_file_download.yml:26]: Why is there an `event_image` reference here?

File: ecms_base/features/custom/ecms_publications/tests/src/Functional/EcmsPublicationsInstallTest.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+namespace Drupal\Tests\ecms_publications\Functional;
+
+// Require the all profiles abstract class since autoloading doesn't work.
+require_once dirname(__FILE__) . '/../../../../../../../tests/src/Functional/AllProfileInstallationTestsAbstract.php';
+
+use Drupal\Tests\ecms_profile\Functional\AllProfileInstallationTestsAbstract;
+
+/**
+ * Class InstallationTest.
```
Comment (pfrilling) [inline ecms_base/features/custom/ecms_publications/tests/src/Functional/EcmsPublicationsInstallTest.php:13]: Nitpick, but this should be `Class EcmsPublicationsInstallTest.`.

**Review by pfrilling — CHANGES_REQUESTED** [review]
One nit-pick and a reference question. Otherwise it looks good.

File: ecms_base/features/custom/ecms_publications/tests/src/Functional/EcmsPublicationsInstallTest.php (hunk 2 above)
Comment (ben-hamelin) [reply ecms_base/features/custom/ecms_publications/tests/src/Functional/EcmsPublicationsInstallTest.php]: Got it!

File: ecms_base/features/custom/ecms_publications/config/install/field.field.node.publication.field_publication_file_download.yml (hunk 1 above)
Comment (ben-hamelin) [reply ecms_base/features/custom/ecms_publications/config/install/field.field.node.publication.field_publication_file_download.yml]: Well that was odd - could not get it to update, so had to manually import the config - which seems to have worked.

**Comment by ben-hamelin** [comment]
Ok once the functional tests are passing again this is ready for your review

**Review by pfrilling — APPROVED** [review]
Looks good once the tests pass.

**Comment by ben-hamelin** [comment]
@pfrilling rebased

**Pull request merged**